### PR TITLE
external_services: add NoNamespace option to the List method

### PR DESCRIPTION
**Pull request by unknwon**
The new `NoNamespace` option allow listing external services under no particular namespace (i.e. site-level external services owned by all site admins). 

This option is then used in handling config overrides of external services from config file. This approach has the minimal change without listing any user-added external services (which are not used in the process of config overrides).

Easier to review by commit.

Part of #12760.

### CODE REVIEW

File: internal/db/external_services.go [hunk 1]
```diff
@@ -57,14 +57,21 @@ type ExternalServiceKind struct {
 
 // ExternalServicesListOptions contains options for listing external services.
 type ExternalServicesListOptions struct {
+	// When true, only include external services not under any namespace (i.e. owned by all site admins),
+	// and value of NamespaceUserID is ignored.
+	NoNamespace bool
```
Comment (tsenart) [inline internal/db/external_services.go:62]: What did consider to choose to have an additional struct field to indicate this predicate? Would the default zero value of `NamespaceUserID` not suffice? Postgres "serial" types always [start at 1](https://www.postgresql.org/docs/9.6/datatype-numeric.html), so we can treat 0 as `NoNamespace`.

Comment (unknwon) [reply internal/db/external_services.go:62]: If we use 0 as no namespace, how do we distinguish between **list all external services** and **list external services with no namespace**?

Comment (tsenart) [reply internal/db/external_services.go:62]: Ah, I see! Well, in that case this is fine! Sorry for misunderstanding. :)

Comment (unknwon) [reply internal/db/external_services.go:62]: No problem at all 😛

**Pull request merged**